Simplenote/Classes/SPNoteEditorViewController.m

```diff
@@ -616,12 +616,14 @@ - (void)updateNote:(Note *)note {
     }
     
     _currentNote = note;
-    [_noteEditorTextView scrollToTop];
-    
-    // push off updating note text in order to speed up animated transition
+    [self.noteEditorTextView scrollToTop];
+
+    // Synchronously set the TextView's contents. Otherwise we risk race conditions with `highlightSearchResults`
+    self.noteEditorTextView.attributedText = [note.content attributedString];
```

Comment:
Is there a potential for this to cause any UI lagginess with large notes?

Reply:
Measured with a really seriously long document, it's in the order of milliseconds. And since it should always run on the main thread, **should not** be an issue.

Thanks for bringing that up!